third-party/linshare/src/main/java/org/apache/james/linshare/client/ByteArrayDecoder.java

```diff
@@ -0,0 +1,44 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.linshare.client;
+
+import java.io.IOException;
+import java.lang.reflect.Type;
+
+import feign.FeignException;
+import feign.Response;
+import feign.codec.Decoder;
+
+public class ByteArrayDecoder implements Decoder {
+    private final Decoder defaultDecoder;
+
+    ByteArrayDecoder(Decoder defaultDecoder) {
+        this.defaultDecoder = defaultDecoder;
+    }
+
+    @Override
+    public Object decode(Response response, Type type) throws IOException, FeignException {
+        if (type.getTypeName().equals("byte[]")) {
+            return new Decoder.Default().decode(response, type);
+        } else {
+            return this.defaultDecoder.decode(response, type);
```

Comment:
You don't need this. No ?

Reply:
I do need it. The problem is we were using a Jackson decoder in the builder for all the rest requests to Linshare. But for downloading something, I don't get a json back, so that decoder doesn't work. The default one though does the job about that. 

But namings as you say in your other comments are confusing clearly, I will improve that.